packages/noya-state/src/reducers/layerReducer.ts

```diff
@@ -242,16 +247,36 @@ export function layerReducer(
         updateSelection(draft.selectedObjects, id, selectionType);
       });
     }
-    case 'addSymbol': {
+    case 'createSymbol': {
       const [, id, name] = action;
       const ids = typeof id === 'string' ? [id] : id;
 
       const page = getCurrentPage(state);
       const pageIndex = getCurrentPageIndex(state);
-      const symbolsPageIndex = getSymbolsPageIndex(state);
 
-      const indexPaths = getIndexPathsForGroup(state, ids);
+      const indexPathsArtboards = getIndexofArboardLayers(state, ids);
+
+      if (indexPathsArtboards.length > 0) {
+        return produce(state, (draft) => {
+          const pages = draft.sketch.pages;
+
+          deleteLayers(indexPathsArtboards, pages[pageIndex]);
+          indexPathsArtboards.forEach((indexPath: IndexPath) => {
+            const artboard = Layers.access(page, indexPath);
+            const symbolMaster = {
+              ...Models.symbolMaster,
+              ...artboard,
+            } as Sketch.SymbolMaster;
+
+            symbolMaster._class = 'symbolMaster';
+            symbolMaster.symbolID = uuid();
```

Comment:
Maybe

```ts
const artboard = Layers.access(page, indexPath) as Sketch.Artboard;

const symbolMaster = {
  ...Models.symbolMaster,
  ...artboard,
  _class: 'symbolMaster' as const,
  symbolID: uuid()
};
```

Reply:
oh the as const ! I was getting an error before